lib/cdn-site-hosting/cdn-site-hosting-construct.ts

```diff
@@ -53,7 +53,9 @@ export class CdnSiteHostingConstruct extends cdk.Construct {
     this.s3Bucket = new s3.Bucket(this, "SiteBucket", {
       bucketName: siteDomain,
       websiteIndexDocument: props.websiteIndexDocument,
-      websiteErrorDocument: props.websiteErrorDocument,
+      websiteErrorDocument: props.isRoutedSpa
+        ? props.websiteErrorDocument
+        : undefined,
```

Comment:
the `ErrorDocument` part of the S3 bucket is now no longer set unless you've opted into SPA routing

Reply:
I'm not sure about this. I think if someone specified `props.websiteErrorDocument` we should honour what they set; if they have omitted it and chosen `props.isRoutedSpa` we could make use `props.websiteIndexDocument` instead.